File: Integraupt-Backend/BackendUsuarios/src/main/java/Controller/EstudianteController.java
package controller;

import dto.EstadoUsuarioForm;
import dto.EstudianteForm;
import java.util.List;
import model.Estudiante;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import service.estudiante.EstudianteService;

@RestController
@RequestMapping("/api/estudiantes")
public class EstudianteController {

    private final EstudianteService service;

    public EstudianteController(EstudianteService service) {
        this.service = service;
    }

    @GetMapping
    public List<Estudiante> listar() {
        return service.listar();
    }

    @GetMapping("/{id}")
    public ResponseEntity<Estudiante> obtener(@PathVariable Integer id) {
        try {
            Estudiante estudiante = service.obtener(id);
            return ResponseEntity.ok(estudiante);
        } catch (RuntimeException e) {
            return ResponseEntity.notFound().build();
        }
    }

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public Estudiante crear(@RequestBody EstudianteForm form) {
        return service.registrar(form);
    }

    @PutMapping("/{id}")
    public ResponseEntity<Estudiante> actualizar(@PathVariable Integer id, @RequestBody EstudianteForm form) {
        try {
            Estudiante estudiante = service.actualizar(id, form);
            return ResponseEntity.ok(estudiante);
        } catch (RuntimeException e) {
            return ResponseEntity.notFound().build();
        }
    }

    @PatchMapping("/{id}/estado")
    public ResponseEntity<Estudiante> actualizarEstado(@PathVariable Integer id, @RequestBody EstadoUsuarioForm form) {
        if (form.activo == null) {
            return ResponseEntity.badRequest().build();
        }
        try {
            Estudiante estudiante = service.actualizarEstado(id, form.activo);
            return ResponseEntity.ok(estudiante);
        } catch (RuntimeException e) {
            return ResponseEntity.notFound().build();
        }
    }

    @DeleteMapping("/{id}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    public void eliminar(@PathVariable Integer id) {
        service.eliminar(id);
    }
}

File: Integraupt-Backend/servicio-usuarios/legacy/controllers/EstudianteController.php
<?php

namespace Controllers;

use Config\App;
use Services\Estudiante\EstudianteService;

class EstudianteController
{
    private EstudianteService $service;

    public function __construct()
    {
        $this->service = new EstudianteService();
    }

    /** GET /api/estudiantes */
    public function listar(): void
    {
        App::json($this->service->listar());
    }

    /** GET /api/estudiantes/{id} */
    public function obtener(int $id): void
    {
        try {
            App::json($this->service->obtener($id));
        } catch (\RuntimeException $e) {
            App::notFound($e->getMessage());
        }
    }

    /** POST /api/estudiantes */
    public function crear(): void
    {
        try {
            $body = App::getBody();
            $estudiante = $this->service->registrar($body);
            App::json($estudiante, 201);
        } catch (\RuntimeException $e) {
            App::error($e->getMessage());
        }
    }

    /** PUT /api/estudiantes/{id} */
    public function actualizar(int $id): void
    {
        try {
            $body = App::getBody();
            App::json($this->service->actualizar($id, $body));
        } catch (\RuntimeException $e) {
            App::error($e->getMessage());
        }
    }

    /** PATCH /api/estudiantes/{id}/estado */
    public function actualizarEstado(int $id): void
    {
        $body = App::getBody();
        if (!isset($body['activo'])) {
            App::error('El campo activo es requerido.');
        }
        try {
            App::json($this->service->actualizarEstado($id, (bool)$body['activo']));
        } catch (\RuntimeException $e) {
            App::notFound($e->getMessage());
        }
    }

    /** DELETE /api/estudiantes/{id} */
    public function eliminar(int $id): void
    {
        try {
            $this->service->eliminar($id);
            http_response_code(204);
        } catch (\RuntimeException $e) {
            App::notFound($e->getMessage());
        }
    }
}
